### ISSUE TITLE
Build failure: libvpx (when cross-compiling for RISCV)

### ISSUE THREAD

**Issue by ToxicFrog** [NixOS/nixpkgs#326129]
### Steps To Reproduce

`nix build 'unstable#pkgsCross.riscv64.libvpx'`

### Build log

```
configure flags: --prefix=/nix/store/31mkq0npgv9qphn3zx1qky3hg68w8bvl-libvpx-riscv64-unknown-linux-gnu-1.14.1 --enable-vp8 --enable-vp8-encoder --enable-vp8-decoder --enable-vp9 --enable-vp9-encoder --enable-vp9-decoder --disable-extra-warnings --disable-werror --disable-install-docs --enable-install-bins --enable-install-libs --disable-install-srcs --disable-debug --disable-gprof--disable-gcov --enable-pic --enable-optimizations --enable-runtime-cpu-detect --disable-thumb --enable-libs --enable-examples --disable-docs --as=yasm --size-limit=5120x3200 --disable-codec-srcs --disable-debug-libs --disable-dequant-tokens --disable-dc-recon --enable-postproc --enable-vp9-postproc --enable-multithread --disable-internal-stats --enable-spatial-resampling --disable-realtime-only --disable-onthefly-bitpacking --disable-error-concealment --enable-shared --disable-small --disable-postproc-visualizer --disable-unit-tests --enable-webm-io --enable-libyuv --disable-decode-perf-tests --disable-encode-perf-tests --disable-multi-res-encoding --enable-temporal-denoising --enable-vp9-temporal-denoising --disable-coefficient-range-checking --enable-vp9-highbitdepth --disable-experimental --target=riscv64-linux-gcc
  enabling vp8
  enabling vp9
  disabling extra_warnings
  [...more enables/disables elided...]
Unrecognized toolchain 'riscv64-linux-gcc'
```

### Additional context

`aarch64-multiplatform` cross compile works in both 24.05 and unstable, so the problem is not cross-compilation in general but the riscv64 build (and possibly other crossbuilds) specifically.

The same build works in 24.05, which means the problem is commit 0af27a5615aedb7ba48496875974fcc4389024f1, "libvpx: fix cross compiling". It's not clear what cross-compilation problems that commit fixes.

### Notify maintainers

@codyopel (maintainer), @RossComputerGuy (author of the breaking commit)

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.37, NixOS, 24.05 (Uakari), 24.05.20240707.1948467`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.0-dev-pre20240619-85f282e`
 - channels(root): `""`
 - nixpkgs: `/etc/channels/nixos`
```

**Comment by RossComputerGuy** [comment]
Fixed by #321361

**Comment by RossComputerGuy** [comment]
To clarify what happened, libvpx never worked with cross compiling. Before the PR which fixed it, we were forcing it the target config which would make it bypass that toolchain check. However that caused issues with aarch64. Right now, libvpx doesn't explicitly support RISC-V so the fix that was just merged told libvpx to use the generic Linux target.
